data_structures/binary_tree/inorder_tree_traversal_2022.py

```diff
@@ -0,0 +1,82 @@
+"""
+Illustrate how to implement inorder traversal in binary search tree.
+Author: Gurneet Singh
+https://www.geeksforgeeks.org/tree-traversals-inorder-preorder-and-postorder/
+"""
+
+
+class BinaryTreeNode:
+    """Defining the structure of BinaryTreeNode"""
+
+    def __init__(self, data: int) -> None:
+        self.data = data
+        self.left_child = None
+        self.right_child = None
+
+
+def insert(node: BinaryTreeNode, new_value: int) -> BinaryTreeNode | None:
+    # if binary search tree is empty, make a new node
+    # and declare it as root
```

Comment:
This is a __doctest__...
```suggestion
    """
    If the binary search tree is empty, make a new node and declare it as root.
    >>> node_a = BinaryTreeNode(12345)
    >>> node_b = insert(node_a, 67890)
    >>> node_a.left_child == node_b.left_child
    True
    >>> node_a.right_child == node_b.right_child
    True
    >>> node_a.data == node_b.data
    False
    """"
```